src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -161,14 +160,18 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(DbConnectionValidat
         return DbConnectionValidatorResult.DocumentTableFound;
     }
 
-    private static SqlBuilder GetSelectBuilderForDocumentTable(SqlBuilder sqlBuilder, string documentTable, string schema)
+    private static string GetDocumentCommandText(ISqlBuilder sqlBuilder, string documentTable, string schema, bool isShellDescriptorDocument = false)
     {
         sqlBuilder.Select();
         sqlBuilder.Selector("*");
         sqlBuilder.Table(documentTable, alias: null, schema);
-        sqlBuilder.Take("1");
 
-        return sqlBuilder;
+        if (isShellDescriptorDocument)
```

Comment:
@hishamco why is this needed? Also why did you remove Take(1) out? Take(1) to improve the query speed since the server will stop looking after it finds the first record.

Reply:
It's needed to check the type value

https://github.com/OrchardCMS/OrchardCore/blob/71fde7c22a098d2003b196b2ad013516e7c1af98/src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs#L185

Reply:
> Also why did you remove Take(1) out?

Oops

Reply:
I already did